src/test/java/ea/maps/MapFactoryTest.java

```diff
@@ -0,0 +1,49 @@
+package ea.maps;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+public class MapFactoryTest {
+
+    private List<City> americaCities;
+
+    @Before
+    public void init() {
+        americaCities = MapFactory.getCityListFromJson("america");
+    }
+
+    @Test
+    public void allCitiesHaveConnectionTest() {
+        americaCities.forEach(c -> assertTrue(c.getConnectedCities().size() > 0));
+    }
+
+    @Test
+    public void allConnectionsCityNamesCorrespondToExistingCity() {
+
+        Set<String> connectionCityNames = americaCities
+            .stream()
+            .flatMap(ac -> ac.getConnectedCities().stream())
+            .map(Connection::getCityName)
+            .collect(Collectors.toSet());
+
+        Set<String> cityNames = americaCities
+            .stream()
+            .map(City::getName)
+            .collect(Collectors.toSet());
+
+        assertThat(cityNames).isEqualTo(connectionCityNames);
+    }
+
+    @Test
+    public void allCitiesHaveThreeDistricts() {
```

Comment:
I think this test is too specific to the implementation of the current map, since not all maps will have this rule. You can remove this test. The other two tests are great though, thank you

Reply:
I'll also change this in one of the commits for my next pull request.